apps/mysql.go

```diff
@@ -0,0 +1,268 @@
+			},
+		},
+	}.Components(tag, uid)
+
+	return c
+}
+
+type LoggingProcessorMysqlSlow struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (LoggingProcessorMysqlSlow) Type() string {
+	return "mysql_slow"
+}
+
+func (p LoggingProcessorMysqlSlow) Components(tag string, uid string) []fluentbit.Component {
+	c := confgenerator.LoggingProcessorParseMultilineRegex{
+		LoggingProcessorParseRegexComplex: confgenerator.LoggingProcessorParseRegexComplex{
+			Parsers: []confgenerator.RegexParser{
+				confgenerator.RegexParser{
+					// Fields documented: https://dev.mysql.com/doc/refman/8.0/en/slow-query-log.html
+					// Sample line: # Time: 2021-10-12T01:13:38.132884Z
+					//              # User@Host: root[root] @ localhost []  Id:    15
+					//              # Query_time: 0.001855  Lock_time: 0.000000 Rows_sent: 0  Rows_examined: 0
+					//              SET timestamp=1634001218;
+					//              SET GLOBAL slow_query_log = 1;
+					// Extra fields w/ low_slow_extra = 'ON'
+					// Sample line: # Time: 2021-10-12T01:34:15.231930Z
+					//              # User@Host: root[root] @ localhost []  Id:    21
+					//              # Query_time: 0.012740  Lock_time: 0.000810 Rows_sent: 327  Rows_examined: 586 Thread_id: 21 Errno: 0 Killed: 0 Bytes_received: 0 Bytes_sent: 41603 Read_first: 2 Read_last: 0 Read_key: 361 Read_next: 361 Read_prev: 0 Read_rnd: 0 Read_rnd_next: 5 Sort_merge_passes: 0 Sort_range_count: 0 Sort_rows: 0 Sort_scan_count: 0 Created_tmp_disk_tables: 0 Created_tmp_tables: 0 Start: 2021-10-12T01:34:15.219190Z End: 2021-10-12T01:34:15.231930Z
+					//              SET timestamp=1634002455;
+					//              select * from information_schema.tables;
+					Regex: `^# Time: (?<time>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d+Z)\s# User@Host:\s+(?<user>[^\[]*)\[(?<database>[^\]]*)\]\s+@\s+((?<host>[^\s]+)\s)?\[(?:(?<ipAddress>[\w\d\.:]+)?)\]\s+Id:\s+(?<tid>\d+)\s+#\s+Query_time:\s+(?<queryTime>[\d\.]+)\s+Lock_time:\s+(?<lockTime>[\d\.]+)\s+Rows_sent:\s+(?<rowsSent>\d+)\s+Rows_examined:\s(?<rowsExamined>\d+)(?:\s+Thread_id:\s+\d+)?(?:\s+Errno:\s(?<errorNumber>\d+))?(?:\s+Killed:\s(?<killed>\d+))?(?:\s+Bytes_received:\s(?<bytesReceived>\d+))?(?:\s+Bytes_sent:\s(?<bytesSent>\d+))?(?:\s+Read_first:\s(?<readFirst>\d+))?(?:\s+Read_last:\s(?<readLast>\d+))?(?:\s+Read_key:\s(?<readKey>\d+))?(?:\s+Read_next:\s(?<readNext>\d+))?(?:\s+Read_prev:\s(?<readPrev>\d+))?(?:\s+Read_rnd:\s(?<readRnd>\d+))?(?:\s+Read_rnd_next:\s(?<readRndNext>\d+))?(?:\s+Sort_merge_passes:\s(?<sortMergePasses>\d+))?(?:\s+Sort_range_count:\s(?<sortRangeCount>\d+))?(?:\s+Sort_rows:\s(?<sortRows>\d+))?(?:\s+Sort_scan_count:\s(?<sortScanCount>\d+))?(?:\s+Created_tmp_disk_tables:\s(?<createdTmpDiskTables>\d+))?(?:\s+Created_tmp_tables:\s(?<createdTmpTables>\d+))?(?:\s+Start:\s(?<startTime>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d+Z))?(?:\s+End:\s(?<endTime>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d+Z))?\s+(?<message>[\s\S]+)`,
```

Comment:
This regex is bordering on unreadable. Do you think you could build it up with code instead? (e.g. a for loop around the list of fields?)

If you don't think that's worth the effort it's also fine with me if you don't change this. I'm just going to have to have faith that the integration tests work.

Reply:
I'd certainly say it's well beyond unreadable. I can look at building the majority of it from a set of fields, and see if that ends up looking cleaner or not.

Reply:
Looks like this has been addressed by https://github.com/GoogleCloudPlatform/ops-agent/pull/219/commits/fbba49c764cefb96223a2d4ec97ece59511db3f3